File: data/multilang_comparison/program_julia/program_julia/src/func46.jl
include("program_julia_head.jl")
function func46(vars::JArray_param, PATH0::UInt64, loopsFactor::Int)::JArray
   pCounter = length(vars.data)
   if (PATH0 & 1) != 0
      params0 = JArray_param{Int}()
      params0.size = 0
      array10 = func47(params0, loopsFactor);
      array10.refC -= 1
      if array10.refC == 0
          empty!(array10.data)
          array10 = nothing
      end
   
   else 
      params0 = JArray_param{Int}()
      params0.size = 0
      array12 = func48(params0, loopsFactor);
      array12.refC -= 1
      if array12.refC == 0
          empty!(array12.data)
          array12 = nothing
      end
   end
   params0 = JArray_param{Int}()
   params0.size = 0
   array14 = func49(params0, loopsFactor);
   return array14;
end

File: data/multilang_comparison/program_cpp_vector/src/func46.cpp
#include "program_cpp_vector.hpp" 
Array* func46(Array_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   if(PATH0 & 1) {
      Array_param params0;
      params0.size = 0;
      std::vector<Array*> data_params0(0);
      params0.data = data_params0;
      Array* array10 = func47(&params0, loopsFactor);
      DEBUG_RETURN(array10->id);
      if(params0.data.size() > 0) {
      	params0.data.clear();
      }
      array10->refC--;
      if(array10->refC == 0) {
         array10->data.clear();
         delete array10;
         DEBUG_FREE(array10->id);
      }
   }
   else {
      Array_param params0;
      params0.size = 0;
      std::vector<Array*> data_params0(0);
      params0.data = data_params0;
      Array* array12 = func48(&params0, loopsFactor);
      DEBUG_RETURN(array12->id);
      if(params0.data.size() > 0) {
      	params0.data.clear();
      }
      array12->refC--;
      if(array12->refC == 0) {
         array12->data.clear();
         delete array12;
         DEBUG_FREE(array12->id);
      }
   }
   Array_param params0;
   params0.size = 0;
   std::vector<Array*> data_params0(0);
   params0.data = data_params0;
   Array* array14 = func49(&params0, loopsFactor);
   DEBUG_RETURN(array14->id);
   if(params0.data.size() > 0) {
   	params0.data.clear();
   }
   return array14;
}

